spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Linear.scala

```diff
@@ -141,20 +141,38 @@ class Linear[T: ClassTag](
 
     if (input.dim() == 1) {
       if (scaleW != 0) {
-        gradWeight.addr(ev.fromType[Double](scaleW), gradOutput, input)
+        if (zeroGradFlag) {
+          gradWeight.addr(ev.zero, gradOutput, ev.fromType[Double](scaleW), input)
+        } else {
+          gradWeight.addr(ev.fromType[Double](scaleW), gradOutput, input)
+        }
       }
 
       if (withBias && scaleB != 0) {
-        gradBias.add(ev.fromType[Double](scaleB), gradOutput)
+        if (zeroGradFlag) {
+          gradBias.mul(gradOutput, ev.fromType[Double](scaleB))
+        } else {
+          gradBias.add(ev.fromType[Double](scaleB), gradOutput)
+        }
       }
     }
     else if (input.dim() == 2) {
       if (scaleW != 0) {
-        gradWeight.addmm(ev.fromType[Double](scaleW), gradOutput.t, input)
+        if (zeroGradFlag) {
+          gradWeight.addmm(ev.zero,
```

Comment:
It seems this operation slows down the backward of Linear @yiheng

Reply:
why, can you dig deeper? These two methods call the same method with only one parameter difference

Reply:
You may need to run some micro-benchmark on this method